Makefile

```diff
@@ -52,6 +60,50 @@ $(foreach variant,$(VARIANTS), \
   ) \
 )
 
+define minor_version
+$(shell echo $(version) | cut -d. -f-2)
+endef
+
+define GEN_R_PATCH_IMAGE_TARGETS
+build-$(version)-$(variant): build-base-$(variant)
+	docker build -t $(BASE_IMAGE):$(version)-$(variant) \
+		--build-arg BASE_IMAGE=$(BASE_IMAGE) \
+		--build-arg R_VERSION=$(version) \
+		$(minor_version)/$(variant)/.
+
+rebuild-$(version)-$(variant): build-base-$(variant)
+	docker build --no-cache -t $(BASE_IMAGE):$(version)-$(variant) \
+		--build-arg BASE_IMAGE=$(BASE_IMAGE) \
+		--build-arg R_VERSION=$(version) \
+		$(minor_version)/$(variant)/.
+
+test-$(version)-$(variant):
+	docker run -it --rm -v $(PWD)/test:/test $(BASE_IMAGE):$(version)-$(variant) bash -l /test/test.sh
+
+bash-$(version)-$(variant):
+	docker run -it --rm -v $(PWD)/test:/test $(BASE_IMAGE):$(version)-$(variant) bash
+
+pull-$(version)-$(variant):
+	docker pull $(BASE_IMAGE):$(version)-$(variant)
+
+push-$(version)-$(variant):
+	docker push $(BASE_IMAGE):$(version)-$(variant)
+
+ifeq (yes,$(INCLUDE_PATCH_VERSIONS))
+BUILD_R_IMAGES += build-$(version)-$(variant)
+REBUILD_R_IMAGES += rebuild-$(version)-$(variant)
+TEST_R_IMAGES += test-$(version)-$(variant)
+PULL_R_IMAGES += pull-$(version)-$(variant)
+PUSH_R_IMAGES += push-$(version)-$(variant)
+endif
+endef
+
+$(foreach variant,$(VARIANTS), \
+  $(foreach version,$(PATCH_VERSIONS), \
+    $(eval $(GEN_R_PATCH_IMAGE_TARGETS)) \
```

Comment:
The patch version builds use the same Dockerfile as the minor versions, but override the R version using a `--build-arg R_VERSION=x.y.z`. This makefile template is pretty the same as the one above but with a few differences:
- adds the extra R version build arg
- excludes the post-push hook that tags the image with the latest patch version
- adds the patch versions to the "all" targets if `INCLUDE_PATCH_VERSIONS=yes`